storm-core/src/jvm/org/apache/storm/StormSubmitter.java

```diff
@@ -233,6 +236,23 @@ public static void submitTopologyAs(String name, Map stormConf, StormTopology to
                 if(topologyNameExists(conf, name, asUser)) {
                     throw new RuntimeException("Topology with name `" + name + "` already exists on cluster");
                 }
+
+                // Dependency uploading only makes sense for distributed mode
+                List<String> jarsBlobKeys;
+                List<String> artifactsBlobKeys;
+
+                DependencyUploader uploader = new DependencyUploader();
+                try {
+                    uploader.init();
+                    jarsBlobKeys = uploadDependencyJarsToBlobStore(uploader);
+                    artifactsBlobKeys = uploadDependencyArtifactsToBlobStore(uploader);
```

Comment:
We can do that for jarsBlobKeys but can't do that for artifactsBlobKeys since we share artifact blobs and we skip uploads if there're same artifact uploaded before.

I'll address this for jarsBlobKeys. Thanks!